src/addrman.cpp

```diff
@@ -1080,10 +1083,9 @@ std::vector<CAddress> AddrManImpl::GetAddr(size_t max_addresses, size_t max_pct,
 {
     LOCK(cs);
     Check();
-    std::vector<CAddress> vAddr;
-    GetAddr_(vAddr, max_addresses, max_pct, network);
+    const auto addresses = GetAddr_(max_addresses, max_pct, network);
```

Comment:
Nit: If the compiler doesn't elide this, the const will force an extra copy: https://stackoverflow.com/questions/25784544/will-returning-a-const-object-from-a-function-prevent-move-construction-from-out/25786015

Is there any specific reason for it?

Reply:
that's so interesting, I hadn't thought about the correlation between copy elision and temporary const variables. thanks for digging in! 

next question is how I get my hands on the spec..

Reply:
It certainly looks like in practice the const doesn't prevent elision: https://godbolt.org/z/Go8xhGP1b